### DISCUSSION TITLE
WebForms port to ASP.NET Core seeks publicity

### DISCUSSION THREAD

**Discussion by simonegli8** [dotnet/aspnetcore#59620] [Ideas]
Hello,

I've ported the WebForms libraries as available in the Reference Source to ASP.NET Core as an OpenSource project on GitHub. Unfortunately my project does not appear on google if one searches for "WebForms for ASP.NET Core" so very few people know about it. I'd be very grateful, if some Microsoft people could write a blog post about my library, so it gets a bit more publicity. The library is just a port of all the System.Web dll's to NET Core, and some middleware to call it from ASP.NET Core. With the aid of this library, one can run ordinary WebForms legacy projects on ASP.NET Core, also on Linux. The library is on GitHub here: https://github.com/webformsforcore/WebFormsForCore. There's also a minimal website about it [here](https://webformsforcore.github.io/). A [video tutorial on how to convert a simple WebForms site to ASP.NET Core can be found here](https://youtu.be/wgg-FziIfNg).
We successfully ported the SolidCP Control Panel, a huge WebForms code base, to ASP.NET Core & Linux with the aid of this library. The goal of running SolidCP on Linux without porting everything to Blazor was also my motivation of creating WebFormsForCore. There is also another similar commercial project CoreForms, but it's license is very expensive. Also there are a few projects on GitHub [WebFormsCore](https://github.com/WebFormsCore/WebFormsCore) & [CoreWebForms](https://github.com/CoreWebForms/CoreWebForms) that don't provide a port of System.Web but only of part of the functionality, but those projects don't allow to port existing WebForms to NET Core without modification.

Greetings & Blessings,

Simon

**Comment by htmlsplash** [comment]
This would be great for the next .net Community Standup. I am curious how you have implemented the solution, what's the interaction between asp.net core and the legacy webforms code.

**Reply by danmoseley** [reply]
@danroth27

**Reply by simonegli8** [reply]
Well I just made System.Web compile with .NET 8 and then I created a Middleware that creates a WorkerRequest from the ASP.NET Core HttpContext and calls the WebForms engine with that WorkerRequest. For the code of the WorkerRequest I looked at the code of the Cassini webserver. Some features are still lacking as for example sharing of authentication & session state between ASP.NET Core and WebForms.

**Reply by simonegli8** [reply]
And how can I apply for a Community Standup? I've seen there's an Issue for EF Core where one can share ideas, but I did not find anything for ASP.NET Core.

**Comment by htmlsplash** [comment]
"Some features are still lacking as for example sharing of authentication & session state between ASP.NET Core and WebForms." - I wonder if you could use System.WebAdapters for this task.

**Comment by lextm** [comment]
Some typical SEO options you might consider:

1. Invite someone with a high rank blog/social account to write about your project.
2. Find top discussions about this topic on Stack Overflow, Reddit, etc. and then post your own answers to promote it. Please mind the rules of those sites, so that your tailored materials won't be banned/removed.

Good luck.
